org.eclipse.xtext/src/org/eclipse/xtext/generator/JavaIoFileSystemAccess.java

```diff
@@ -129,7 +130,9 @@ protected void generateTrace(String generatedFile, String outputConfigName, Char
 					traceSerializer.writeTraceRegionTo(traceRegion, out);
 					if(callBack != null) 
 						callBack.fileAdded(traceFile);
-				} finally {
+				} catch (TraceNotFoundException e) {
```

Comment:
`getTraceRegion()` should be called before the output stream is created in order to avoid opening the trace file unnecessarily.